.azure/pipelines/jobs/project-ci.yml

```diff
@@ -13,37 +13,61 @@
 #       The matrix of configurations to run. By default, it runs a Debug and Release build on all platforms
 #   codeSign: boolean
 #       This build definition is enabled for code signing. (Only applies to Windows)
+#   runCodeQL3000: boolean
+#       This build should run CodeQL3000 instead of the regular build
 
 parameters:
   buildArgs: ''
   beforeBuild: []
   afterBuild: []
   codeSign: false
   variables: {}
+  runCodeQL3000: false
 
 jobs:
-- template: default-build.yml
-  parameters:
-    agentOs: Windows
-    matrix: ${{ parameters.matrix }}
-    buildArgs: ${{ parameters.buildArgs }}
-    beforeBuild: ${{ parameters.beforeBuild }}
-    afterBuild: ${{ parameters.afterBuild }}
-    codeSign: ${{ parameters.codeSign }}
-    variables: ${{ parameters.variables }}
-- template: default-build.yml
-  parameters:
-    agentOs: macOS
-    matrix: ${{ parameters.matrix }}
-    buildArgs: ${{ parameters.buildArgs }}
-    beforeBuild: ${{ parameters.beforeBuild }}
-    afterBuild: ${{ parameters.afterBuild }}
-    variables: ${{ parameters.variables }}
-- template: default-build.yml
-  parameters:
-    agentOs: Linux
-    matrix: ${{ parameters.matrix }}
-    buildArgs: ${{ parameters.buildArgs }}
-    beforeBuild: ${{ parameters.beforeBuild }}
-    afterBuild: ${{ parameters.afterBuild }}
-    variables: ${{ parameters.variables }}
+- ${{ if and(ne(variables['System.TeamProject'], 'public'), eq(parameters.runCodeQL3000, 'true')) }}:
+  - template: default-build.yml
+    parameters:
+      agentOs: Windows
+      matrix: ${{ parameters.matrix }}
+      buildArgs: ${{ parameters.buildArgs }}
+      beforeBuild:
+      - script: "echo ##vso[build.addbuildtag]CodeQL3000"
+        displayName: 'Set CI CodeQL3000 tag'
+      - task: CodeQL3000Init@0
+        displayName: CodeQL Initialize
+      afterBuild:
+      - task: CodeQL3000Finalize@0
+        displayName: CodeQL Finalize
+      codeSign: false
+      variables: ${{ parameters.variables }}
+      runCodeQL3000: true
+- ${{ else }}: # regular build
+  - template: default-build.yml
+    parameters:
+      agentOs: Windows
+      matrix: ${{ parameters.matrix }}
+      buildArgs: ${{ parameters.buildArgs }}
+      beforeBuild: ${{ parameters.beforeBuild }}
+      afterBuild: ${{ parameters.afterBuild }}
+      codeSign: ${{ parameters.codeSign }}
+      variables: ${{ parameters.variables }}
+      runCodeQL3000: false
```

Comment:
default-build.yml doesn't have a parameter w/ this name. even if it did, no reason to set parameters to their defaults